app/Platform/Controllers/RelatedGamesTableController.php

```diff
@@ -54,23 +58,53 @@ public function __invoke(Request $request): View
             });
         }
 
+        if (isset($filterOptions['status'])) {
+            $this->gameListService->filterGameList(function ($game) use ($filterOptions) {
+                return $this->gameListService->useGameStatusFilter($game, $filterOptions['status']);
+            });
+        }
+
         $this->gameListService->mergeWantToPlay($loggedInUser);
 
         $this->gameListService->sortGameList($sortOrder);
 
         $availableSorts = $this->gameListService->getAvailableSorts();
-        $availableFilters = [
+        $availableCheckboxFilters = [
             'console' => 'Group by console',
             'populated' => 'Only with achievements',
         ];
+        $availableSelectFilters = [
+            [
+                'kind' => 'status',
+                'label' => 'Status',
+                'options' => [
+                    'all' => 'All games',
+                    'unstarted' => 'No progress',
+                    'unawarded' => 'Unfinished',
+                    'eq-beaten-softcore' => 'Beaten (softcore)',
+                    'eq-beaten-hardcore' => 'Beaten',
+                    'eq-completed' => 'Completed',
+                    'eq-mastered' => 'Mastered',
+                    'any-beaten' => 'Beaten, either hardcore or softcore',
+                    'gte-completed' => 'Completed or mastered',
+                    'awarded' => 'Games with any award',
+                    'eq-revised' => 'Games with awards for revised sets',
+                    'gte-beaten-softcore' => 'Games with softcore awards',
+                    'gte-beaten-hardcore' => 'Games with hardcore awards',
+                    'any-softcore' => 'Games with any softcore progress',
```

Comment:
This and `any-hardcore` throw an exception:
> Trying to access array offset on value of type null

Reply:
`any-hardcore` removed entirely in latest, however `any-softcore` has been retained and the error has been fixed.